### DISCUSSION TITLE
What takes priority, Cron schedule and trigger start time ?

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#2239] [Q&A]
I'm setting up triggers with cron jobs to run at a specific time. I'm wondering what happens if I set a start time on the trigger as well.
IF the start date and time is set after the cron schedule on creation does it skip that cron time?

Is there a way to get the cron schedule time so that they I can pass it a date and it grabs the cron time?

Now sure how to handle having a start time with a cron schedule.

**Comment by lahma** [comment]
Triggers calculate next fire time from the start time, so trigger will fire earliest the start time.

> Is there a way to get the cron schedule time so that they I can pass it a date and it grabs the cron time?

This question I don't quite understand.

**Reply by CrazyMoka** [reply]
say its currently 11am. If I set a Cron to run every day at 6am and I set a StartAt for 8am, will it run at 8am then the next day at 6am moving forward or will it run at 6am then 8am then 6am again moving forward?

**Reply by lahma** [reply]
To answer this question more generally, I would suggest you look at `TriggerUtils` class and its helpers like `ComputeFireTimes` and you can just output the next fire times Quartz will follow.
